File: src/main/java/gargoyle/xenox/i/AudioManager.java
package gargoyle.xenox.i;

import gargoyle.xenox.util.res.audio.AudioClip;

import java.net.URL;

public interface AudioManager {
    void musicLoop();

    void musicPlay();

    void musicSet(AudioClip clip);

    void musicSet(URL... clip);

    void musicStop();

    void soundPlay(AudioClip clip);

    void soundPlay(URL... clip);
}

File: src/main/kotlin/gargoyle/xenox/i/AudioManager.kt
package gargoyle.xenox.i

import gargoyle.xenox.util.res.audio.AudioClip
import java.net.URL

interface AudioManager {
    fun musicLoop()
    fun musicPlay()
    fun musicSet(clip: AudioClip)
    fun musicSet(vararg clip: URL)
    fun musicStop()
    fun soundPlay(clip: AudioClip)
    fun soundPlay(vararg clip: URL)
}
